wagtail/models/i18n.py

```diff
@@ -174,13 +174,143 @@ def is_active(self) -> bool:
             return self.is_default
 
 
+class TranslatableQuerySetMixin:
+    """
+    QuerySet mixin for translatable models.
+
+    This mixin provides methods for query sets of translatable models. If your
+    translatable model inherits from
+    :py:class:`TranslatableMixin <wagtail.models.TranslatableMixin>` and defines a
+    custom manager that inherits from ``models.QuerySet``, be sure to also inherit
+    from this queryset mixin to retain the features of this mixin.
+
+    .. code-block:: python
+
+        class MyTranslatableModelQuerySet(TranslatableQuerySetMixin, models.QuerySet):
+            pass
+
+        class MyTranslatableModel(TranslatableMixin):
+            objects = MyTranslatableModelQuerySet.as_manager()
+
+    If your translatable model does not define a custom manager ``objects``, you won't
+    need this mixin. The ``TranslatableMixin`` already provides a manager that inherits
+    from this mixin.
+    """
+
+    def localized(
+        self: models.QuerySet,
+        keep_order: bool = False,
+        include_draft_translations: bool = False,
+    ):
+        """
+        Return a localized version of this queryset.
+
+        Localize means to check, for every instance in the queryset, if there is a
+        translation into the active locale. If a translation exists, use the translated
+        instance, otherwise use the instance itself. The result is a queryset of the
+        same length as the original queryset.
```

Comment:
A localized queryset is one where objects are replaced (where possible) with versions translated into the active locale.